src/super_gradients/training/processing/processing.py

```diff
@@ -357,6 +371,162 @@ def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, None]:
         return cropped_image, None
 
 
+@register_processing(Processings.SegResizeWithPadding)
+class SegResizeWithPadding(Processing, ABC):
+    """Resize image to given image dimensions while preserving aspect ratio (padding might be used).
+
+    :param output_shape:    (H, W)
+    :param pad_value:    padding value (will be used if padding needed)
+    """
+
+    def __init__(self, output_shape: Tuple[int, int], pad_value: int):
+        self.output_shape = output_shape
+        self.pad_value = pad_value
+
+    def preprocess_image(self, image: np.ndarray) -> Tuple[np.ndarray, SegmentationRescaleWithPaddingMetadata]:
+        height, width = image.shape[:2]
+        scale_factor = min(self.output_shape[0] / height, self.output_shape[1] / width)
+
+        if scale_factor != 1.0:
+            new_height, new_width = round(height * scale_factor), round(width * scale_factor)
+            image = _rescale_image(image, target_shape=(new_height, new_width))
+
+        padding_coordinates = _get_center_padding_coordinates(input_shape=image.shape, output_shape=self.output_shape)
+        processed_image = _pad_image(image=image, padding_coordinates=padding_coordinates, pad_value=self.pad_value)
+
+        return processed_image, SegmentationRescaleWithPaddingMetadata(
+            original_shape=(height, width), scale_factor=scale_factor, padding_coordinates=padding_coordinates
+        )
+
+    def postprocess_predictions(self, predictions: SegmentationPrediction, metadata: SegmentationRescaleWithPaddingMetadata) -> SegmentationPrediction:
+        predictions.segmentation_map = predictions.segmentation_map[
+            metadata.padding_coordinates.top : predictions.segmentation_map_shape[0] - metadata.padding_coordinates.bottom,
+            metadata.padding_coordinates.left : predictions.segmentation_map_shape[1] - metadata.padding_coordinates.right,
+        ]
+        predictions.segmentation_map = _rescale_image(predictions.segmentation_map.astype(np.uint8), target_shape=metadata.original_shape)
+        predictions.segmentation_map = predictions.segmentation_map.astype(np.int64)
+        return predictions
+
+
+@register_processing(Processings.SegmentationRescale)
+class SegmentationRescale(Processing, ABC):
```

Comment:
Why ABC? Is that a mistake ?

Reply:
fixed (b55fcd97f9cf830a80fe7a9ed00aec6999fb374a)